src/components/ItemList/ItemList.tsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import { EMPTY_ITEMS_ARRAY } from './data';
+import { IBeerDetails } from './data';
+import API from '../API/API';
+
+type PropsType = { searchString: string | null };
+
+class ItemList extends React.PureComponent<
+  PropsType,
+  { beerList: IBeerDetails[]; isLoading: boolean; requestOK: boolean }
+> {
+  constructor(props: PropsType) {
+    super(props);
+
+    this.state = {
+      beerList: EMPTY_ITEMS_ARRAY,
+      isLoading: false,
+      requestOK: true,
+    };
+
+    this.queryData = this.queryData.bind(this);
+  }
+
+  private async queryData(): Promise<void> {
+    this.setState({ isLoading: true });
+
+    const dataQuery = new API();
+    this.setState({
+      requestOK: await dataQuery.queryItems(
+        this.props.searchString,
+        (obj: { beerList: IBeerDetails[] }) => this.setState(obj)
+      ),
+    });
+
+    this.setState({ isLoading: false });
+  }
+
+  public componentDidMount(): void {
+    this.queryData();
+  }
+
+  public componentDidUpdate(prevProps: PropsType) {
+    if (this.props.searchString !== prevProps.searchString) {
+      this.queryData();
+    }
+  }
+
+  public render(): React.ReactNode {
+    return (
+      <div>
+        <div>
+          {!this.state.beerList.length || !this.state.requestOK ? (

```

Comment:
Please, try to avoid "logical not" operator in component markup. It's difficult to read. (this is just an advice)

Reply:
Fixed in https://github.com/ylakhtin/rs-react-components/commit/0b1b88c08b4feefd436a58416b2ffcc44a1bc92e